terraform/resource.go

```diff
@@ -316,7 +316,8 @@ func (c *ResourceConfig) get(
 			// prefix so were split as path components above.
 			actualKey := strings.Join(parts[i-1:], ".")
 			if prevMap, ok := previous.(map[string]interface{}); ok {
-				return prevMap[actualKey], true
+				v, ok := prevMap[actualKey]
+				return v, ok
```

Comment:
In addition to the above test we should be able to reproduce this with a `ResourceConfig.Get` test, I'd feel much safer if we did that in addition to the above.